### PR TITLE
Add back system test for AWS auth manager

### PR DESCRIPTION
**Pull request by vincbeck**
#37947 has been reverted because it was making fail all AWS system tests. This is because the constant `TEST_FILE_IDENTIFIER` was wrongly set. This PR include the revert of the revert commit to add back the system test and the fix to not make the AWS system tests fail.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by o-nikolas** [comment]
Did you get a passing local run of one or more of the system tests that failed with the last PR?

**Comment by vincbeck** [comment]
> Did you get a passing local run of one or more of the system tests that failed with the last PR?

Yes

**Comment by ferruzzi** [comment]
Interesting, can you ELI5 why this would break only some of the tests and not others?

**Comment by vincbeck** [comment]
The function `_get_test_name` in `tests/system/providers/amazon/aws/utils/__init__.py` return the test file name by looking at the files from `inspect.stack()`. My understanding is `inspect.stack()` returns all files loaded in memory (thus, the test file and all the imports). In order to find the test file, we are using `TEST_FILE_IDENTIFIER` to check whether the file is named like a test file (example_...py or test_...py). In #37947 I added "test" as pattern to check whether a given file is a test file. The problem with "test" is it is too permissive. Basically any files under `/tests` matches it. This is the issue.

The consequence is, any system test which need to fetch some external parameter (using SSM) would fail. That explains why some system tests kept succeeding. The reason why these system test failed is `_get_test_name` was not returning the correct system test file name but any random file which than lead to "Could not find the parameter in SSM" because the system test was not providing the correct system test name.

**Review by ferruzzi — APPROVED** [review]
Thanks for the explanation.
